### PR TITLE
22Q2-34 ✅ EKS upgrade

### PR DESCRIPTION
**Pull request by yngvark**
This adds an upgrade script together with a README for how to bump EKS.

## Description

Strategy:
* The upgrade script can be used to bump to any EKS version.
* If a specific version of EKS needs some extra stuff, this can be handled by a sub section in the README targeting those specifics (for instance updating Ingress version, or requiring a minimum okctl version)

## Motivation and Context

[22Q2-34](https://trello.com/c/ou4TwjcP/34-lage-standalone-upgrade-for-%C3%A5-oppgradere-eks).

## How to prove the effect of this PR?

* Create a cluster with `okctl apply cluster`, preferrably version 1.20. Or use an existing one.
* Run through the README (and hence upgrade.sh) to upgrade it to 1.21.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: gists/bump-eks/README.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+```
+brew install watch
+```
+
+# Step 2: Log in to the environment
+
+Set `AWS_PROFILE` to the correct AWS profile from `~/.aws/config`. If you have not set this up, have a look at [Authenticating to AWS](https://www.okctl.io/authenticating-to-aws/#aws-single-sign-on-sso).
+
+```sh
+export AWS_PROFILE=my-dev-account
```
Comment (deifyed) [inline gists/bump-eks/README.md:91]: Maybe use "relevant account" instead of "my dev-account"

File: gists/bump-eks/README.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+
+
+
+
+# Step 4: Adapt to EKS version specific requirements
+
+## EKS 1.21
+
+Make sure you have upgraded to Okctl version TODO or later.
```
Comment (deifyed) [inline gists/bump-eks/README.md:109]: version TODO?

File: gists/bump-eks/README.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+TODO: Okctl doesn't support EKS 1.21. We will update this guide when it is.
+
+## EKS 1.22
+
+If you're upgradring to EKS 1.22, you must follow the below steps. If you do not, your application **probably will stop working**.
```
Comment (deifyed) [inline gists/bump-eks/README.md:115]:
```suggestion
If you're upgrading to EKS 1.22, you must follow the below steps. If you do not, your application **probably will stop working**.
```

File: gists/bump-eks/README.md [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+$ grep -nRsH "apiVersion: networking.k8s.io/v1beta1"
+--
+infrastructure/applications/okctl-reference-app/base/ingress.yaml:1:apiVersion: networking.k8s.io/v1beta1
+infrastructure/applications/okctl-reference-app/base/ingress.yaml-2-kind: Ingress
+--
+infrastructure/applications/hello/base/ingress.yaml:1:apiVersion: networking.k8s.io/v1beta1
+infrastructure/applications/hello/base/ingress.yaml-2-kind: Ingress
+```
+
+For every file in the result, edit it and replace the `apiVersion` so it becomes like this:
```
Comment (deifyed) [inline gists/bump-eks/README.md:146]: Isn't there a change in how you describe routes between `extensions/v1beta1` and `networking.k8s.io/v1`

**Review by deifyed — APPROVED** [review]
Next level bash scripting

File: gists/bump-eks/README.md (hunk 4 above)
Comment (yngvark) [reply gists/bump-eks/README.md:144]: True, good catch. I'll add it.

**Pull request merged**